R/SoilProfileCollection-methods.R

```diff
@@ -80,12 +80,41 @@ setMethod(
     }
 
     # filter out missing data
-    h <- .data.frame.j(h, target.names, aqp_df_class(x))
+    h <- aqp:::.data.frame.j(h, target.names, aqp_df_class(x))
     h <- h[complete.cases(h),]
 
     # compute max depth within each profile
-    d <- tapply(h[[hz_bottom_depths]], h[[idname(x)]], max, na.rm = TRUE)
-
+    if (aqp_df_class(x) == "data.table" & 
+        requireNamespace("data.table") ) {
+      
+      # base R faster for big data with no existing key
+      # but if the key is already set, then this is ~2x faster with 1M profiles (sorted numeric IDs)
+      if (idname(x) %in% data.table::key(h)) {
+        idn <- idname(x)
+        # # with no key
+        # user  system elapsed 
+        # 7.26    5.52   16.83 
+        # # with pre-set key
+        # user  system elapsed 
+        # 2.07    0.00    2.08  
+        
+        # cant invoke this for something like min/max probably -- might do better on linux
+        # data.table::setkeyv(h, c(idn))
+        
+        dep <- h[[hz_bottom_depths]]
+        d <- dep[h[, .I[hz_bottom_depths == max(hz_bottom_depths, na.rm = T)][1],
+                     by = idn]$V1] 
+        
+        # return from here for data.table
+        return(min(d, na.rm = TRUE))
+      }
+    }
```

Comment:
correct, because of the only modest improvement for data.table -- it is not outweighed by setting a key -- I don't think

It was something like 5-6 seconds for tapply, versus 2 seconds for data.table keyed join, versus like 16 seconds for setting the key + doing the join